### PR TITLE
Update guidance on ARIA `role` attribute

### PR DESCRIPTION
**Pull request by Malvoz**
This PR adds an example of "bad" vs "good" markup, and a link to https://www.w3.org/TR/html-aria/#implicit such that developers can find the corresponding ARIA role if they want to convey the semantics of a certain HTML element.

<hr>

As an aside, I _wanted_ to include this table to provide a shortcut for which `role` is implicit of each (widely used) element. 

| Element      | Implicit ARIA `role`
| ------------ | ------------------- |
| `<header>`   | `banner`            |
| `<main>`     | `main`              |
| `<footer>`   | `contentinfo`       |
| `<article>`  | `article`           |
| `<section>`  | `region`            |
| `<aside>`    | `complementary`     |

But the implicitness may change for certain elements depending on some factors, e.g. the [`<header>` element](https://www.w3.org/TR/html-aria/#el-header):

> If not a descendant of an `article`, `aside`, `main`, `nav` or `section` element, or an element with `role=article`, `complementary`, `main`, `navigation` or `region` then `role=banner`, otherwise No corresponding role

And since I didn't want to get too wordy, I decided not to include it. Developers can use the link instead and find out for themselves what to use.

### CODE REVIEW

**Comment by hteumeuleu** [comment]
Thank you for this PR. This is a good idea. Could you just make sure that only the corresponding paragraph is changed? (and not the indentation of the code in the _tables for layout_ section)

**Comment by Malvoz** [comment]
Alright, done!

**Comment by hteumeuleu** [comment]
Thanks!
